File: rust/src/models/usage_fee_structure.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::{billable_metric_id::BillableMetricId, usage_model_enum::UsageModelEnum};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct UsageFeeStructure {
    pub metric_id: BillableMetricId,

    pub model: UsageModelEnum,
}

impl UsageFeeStructure {
    pub fn new(metric_id: BillableMetricId, model: UsageModelEnum) -> Self {
        Self { metric_id, model }
    }
}

File: typescript/src/models/usageFeeStructure.ts
// this file is @generated
import { type BillableMetricId, BillableMetricIdSerializer } from "./billableMetricId";
import { type UsageModelEnum, UsageModelEnumSerializer } from "./usageModelEnum";

export interface UsageFeeStructure {
  metricId: BillableMetricId;

  model: UsageModelEnum;
}

export const UsageFeeStructureSerializer = {
  _fromJsonObject(object: any): UsageFeeStructure {
    return {
      metricId: BillableMetricIdSerializer._fromJsonObject(object["metric_id"]),
      model: UsageModelEnumSerializer._fromJsonObject(object["model"]),
    };
  },

  _toJsonObject(self: UsageFeeStructure): any {
    return {
      metric_id: BillableMetricIdSerializer._toJsonObject(self.metricId),
      model: UsageModelEnumSerializer._toJsonObject(self.model),
    };
  },
};
